code/modules/recycling/conveyor2.dm

```diff
@@ -125,43 +124,39 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		return FALSE
 	return TRUE
 
-// returns true if there is at least 1 item on the belt that can be moved
-/obj/machinery/conveyor/proc/check_belt_contents()
-	for(var/atom/movable/AM in loc)
-		if(!AM.anchored)
-			return TRUE
-	return FALSE
+// machine process
+// move items to the target location
+/obj/machinery/conveyor/process()
+	if(machine_stat & (BROKEN | NOPOWER))
+		return
+
+	//If the conveyor is broken or already moving items
+	if(!operating || conveying)
+		return
 
-// populates `affecting` with the first 30 items in the belt's contents
-/obj/machinery/conveyor/proc/get_belt_contents()
+	use_power(6)
+
+	//get the first 30 items in contents
 	affecting = list()
 	var/i = 0
-	for(var/atom/movable/AM in loc)
-		if(AM.anchored) // If it can't be moved, ignore it
+	for(var/item in loc.contents)
+		if(item == src)
 			continue
```

Comment:
Try this
```suggestion
	for(var/item in loc.contents-src)
```
Drip of unload

Reply:
Yeah you're right I should fix that while I'm here. Doing `if(item == src)` is gross.